analyses/cell-type-ewings/cell_type_annotations/SCPCS000490/SCPCL000822_tumor-cell-validation.Rmd

```diff
@@ -0,0 +1,492 @@
+```
+
+```{r}
+# Input files
+sce_file <- file.path(sample_dir, "SCPCL000822_processed.rds")
+marker_genes <- file.path(module_base, "references", "tumor-marker-genes.tsv")
+
+# results from annotation workflow 
+results_dir <- file.path(module_base, "results", "annotate_tumor_cells_output", "SCPCS000490")
+
+marker_gene_results_file <- file.path(results_dir, "SCPCL000822_tumor-normal-classifications.tsv")
+copykat_predictions_file <- file.path(results_dir, "SCPCL000822_copykat-classifications.tsv")
+
+infercnv_predictions_file <- file.path(results_dir, "SCPCL000822_infercnv-classifications.tsv")
+infercnv_metadata_file <- file.path(results_dir, "infercnv", "SCPCL000822_cnv-metadata.tsv")
+
+geneset_scores_file <- file.path(results_dir, "SCPCL000822_gene-set-scores.tsv")
+
+# output files 
+final_annotations_dir <- file.path(module_base, "cell_type_annotations", "SCPCS000490")
+fs::dir_create(final_annotations_dir)
+final_annotations_file <- file.path(final_annotations_dir, "SCPCL000822_tumor-classifications.tsv")
+```
+
+
+```{r}
+# read in all files 
+sce <- readr::read_rds(sce_file)
+
+# read in marker genes table 
+marker_genes_df <- readr::read_tsv(marker_genes) |> 
+  # account for genes being from multiple sources
+  dplyr::select(cell_type, ensembl_gene_id, gene_symbol) |> 
+  dplyr::distinct()
+
+marker_gene_results_df <- readr::read_tsv(marker_gene_results_file)
+
+# copykat output
+copykat_results_df <- readr::read_tsv(copykat_predictions_file) |> 
+  # only select no_ref classification and rename to be consistent with naming of other classifications
+  dplyr::select(barcodes, copykat_classification = no_ref) |> 
+  dplyr::mutate(
+    copykat_classification = dplyr::case_when(
+      copykat_classification == "aneuploid" ~ "Tumor",
+      copykat_classification == "diploid" ~ "Normal",
+      TRUE ~ copykat_classification
+    )
+  )
+
+# infercnv output 
+infercnv_results_df <- readr::read_tsv(infercnv_predictions_file)
+infercnv_metadata_df <- read.delim(infercnv_metadata_file) |> 
+  tibble::rownames_to_column("barcodes")
+
+# remove MT and GL chr, comments in docs recommend that these are not useful and can be inaccurate 
+chr_columns_to_keep <- which(!stringr::str_detect(names(infercnv_metadata_df), "chrMT|chrGL"))
+infercnv_metadata_df <- infercnv_metadata_df[, chr_columns_to_keep]
+
+# gene set scores 
+geneset_scores_df <- readr::read_tsv(geneset_scores_file)
+```
+
+```{r}
+# grab all coldata and add in all classifications and results to use for plotting throughout notebook 
+# get list of marker genes to use 
+marker_genes <- marker_genes_df |> 
+  dplyr::filter(cell_type == "tumor") |> 
+  dplyr::pull(ensembl_gene_id)
+
+# get the gene expression counts for all marker genes
+marker_gene_exp <- logcounts(sce[marker_genes, ]) |>
+  as.matrix() |>
+  t() |>
+  as.data.frame() |>
+  tibble::rownames_to_column("barcodes")
+
+# pull out the UMAP coordinates and make a data frame to use for plotting
+coldata_df <- sce |> 
```

Comment:
Ah, this (the pivot after join) explains why there are duplicates below. I was confused by calling this `coldata_df` when it is not really that after faceting. I might give it a more explicit name that reflects that it has marker expression values as well.